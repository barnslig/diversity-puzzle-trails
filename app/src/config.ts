interface Config {
  /**
   * API endpoints
   */
  apiEndpoints: {
    clock: (gameId: string) => string;
    code: (codeId: string, gameId: string) => string;
    parameters: (gameId: string) => string;
  };

  /**
   * Allowed URL origins for QR codes that are recognized by the scanner
   */
  allowedCodeOrigins: string[];
}

const config: Config = {
  apiEndpoints: {
    clock: (gameId: string) => `${process.env.API_ROOT}/games/${gameId}/clock`,
    code: (codeId: string, gameId: string) =>
      `${process.env.API_ROOT}/games/${gameId}/codes/${codeId}`,
    parameters: (gameId: string) =>
      `${process.env.API_ROOT}/games/${gameId}/parameters`,
  },
  allowedCodeOrigins: [window.location.origin, "https://abc-dpt.netlify.app"],
};

export default config;
