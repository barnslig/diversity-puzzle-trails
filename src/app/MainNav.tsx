import {
  Paper,
  BottomNavigation,
  BottomNavigationAction,
  makeStyles,
} from "@material-ui/core";
import { Home } from "@material-ui/icons";
import * as React from "react";
import { FormattedMessage } from "react-intl";
import { useLocation } from "wouter";
import QrCodeScanner from "../common/icons/QrCodeScanner";

const useStyles = makeStyles((theme) => ({
  bottomNavigation: {
    bottom: 0,
    left: 0,
    position: "fixed",
    width: "100%",
    zIndex: theme.zIndex.appBar,
  },
}));

type MainNavProps = {};

const MainNav = (props: MainNavProps) => {
  const classes = useStyles();
  const [location, setLocation] = useLocation();

  return (
    <Paper elevation={8} component="nav" className={classes.bottomNavigation}>
      <BottomNavigation
        value={location}
        onChange={(ev, value) => value !== location && setLocation(value)}
        showLabels
      >
        <BottomNavigationAction
          value="/"
          label={
            <FormattedMessage
              defaultMessage="Status"
              description="main nav main page item label"
            />
          }
          icon={<Home />}
        />
        <BottomNavigationAction
          value="/scan"
          label={
            <FormattedMessage
              defaultMessage="Scanner"
              description="main nav scanner page item label"
            />
          }
          icon={<QrCodeScanner />}
        />
      </BottomNavigation>
    </Paper>
  );
};

export default MainNav;
