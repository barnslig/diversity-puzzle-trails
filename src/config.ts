interface Config {
  /**
   * Allowed URL origins for QR codes that are recognized by the scanner
   */
  allowedCodeOrigins: string[];
}

const config: Config = {
  allowedCodeOrigins: [window.location.origin],
};

export default config;
