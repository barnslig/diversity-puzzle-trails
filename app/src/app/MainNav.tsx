import { ChatBubble, Home } from "@mui/icons-material";
import { FormattedMessage } from "react-intl";
import { Paper, BottomNavigation, BottomNavigationAction } from "@mui/material";
import { useLocation } from "wouter";
import * as React from "react";

import QrCodeScanner from "../common/icons/QrCodeScanner";

import config from "../config";

type MainNavProps = {};

const MainNav = (props: MainNavProps) => {
  const [location, setLocation] = useLocation();

  return (
    <Paper
      elevation={8}
      component="nav"
      sx={{
        bottom: 0,
        left: 0,
        position: "fixed",
        width: "100%",
        zIndex: "appBar",
      }}
    >
      <BottomNavigation
        value={location}
        onChange={(ev, value) => value !== location && setLocation(value)}
        showLabels
      >
        <BottomNavigationAction
          value="/"
          label={
            <FormattedMessage
              defaultMessage="Status"
              description="main nav main page item label"
            />
          }
          icon={<Home />}
        />
        <BottomNavigationAction
          value="/scan"
          label={
            <FormattedMessage
              defaultMessage="Scanner"
              description="main nav scanner page item label"
            />
          }
          icon={<QrCodeScanner />}
        />
        {config.featureMessages && (
          <BottomNavigationAction
            value="/messages"
            label={
              <FormattedMessage
                defaultMessage="Nachrichten"
                description="main nav messages page item label"
              />
            }
            icon={<ChatBubble />}
          />
        )}
      </BottomNavigation>
    </Paper>
  );
};

export default MainNav;
