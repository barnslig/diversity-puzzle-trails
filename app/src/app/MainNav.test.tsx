import { screen } from "@testing-library/react";
import * as React from "react";
import userEvent from "@testing-library/user-event";

import config from "../config";
import MainNav from "./MainNav";
import render from "../common/testing/render";

it("renders correctly", () => {
  const { container } = render(<MainNav />);
  expect(container.firstChild).toMatchSnapshot();
});

it("sets the location on click", () => {
  render(<MainNav />);

  expect(window.location.pathname).toEqual("/");

  if (config.featureMessages) {
    userEvent.click(screen.getByText(/Nachrichten/));
    expect(window.location.pathname).toEqual("/messages");
  }

  userEvent.click(screen.getByText(/Scanner/));
  expect(window.location.pathname).toEqual("/scan");

  userEvent.click(screen.getByText(/Status/));
  expect(window.location.pathname).toEqual("/");
});
