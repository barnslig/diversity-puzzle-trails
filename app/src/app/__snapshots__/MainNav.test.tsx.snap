// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 css-avw9ge-MuiPaper-root"
>
  <div
    class="MuiBottomNavigation-root css-16lloyr-MuiBottomNavigation-root"
  >
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="HomeIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
      <span
        class="MuiBottomNavigationAction-label Mui-selected css-imwso6-MuiBottomNavigationAction-label"
      >
        Status
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9.5 6.5v3h-3v-3h3M11 5H5v6h6V5zm-1.5 9.5v3h-3v-3h3M11 13H5v6h6v-6zm6.5-6.5v3h-3v-3h3M19 5h-6v6h6V5zm-6 8h1.5v1.5H13V13zm1.5 1.5H16V16h-1.5v-1.5zM16 13h1.5v1.5H16V13zm-3 3h1.5v1.5H13V16zm1.5 1.5H16V19h-1.5v-1.5zM16 16h1.5v1.5H16V16zm1.5-1.5H19V16h-1.5v-1.5zm0 3H19V19h-1.5v-1.5zM22 7h-2V4h-3V2h5v5zm0 15v-5h-2v3h-3v2h5zM2 22h5v-2H4v-3H2v5zM2 2v5h2V4h3V2H2z"
        />
      </svg>
      <span
        class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
      >
        Scanner
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ChatBubbleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
      >
        Nachrichten
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</nav>
`;
